### PR TITLE
Port some X-Ray Daemon configs

### LINKED ISSUE #335: Port a subset of the existing X-Ray Daemon configurations into the OT X-Ray receiver
Not all the X-Ray daemon [configurations](https://docs.aws.amazon.com/xray/latest/devguide/xray-daemon-configuration.html#xray-daemon-configuration-configfile) make sense for the OT X-Ray receiver. We will only port a selected subset.

### PR DESCRIPTION
**Pull request by hencrice**
**Description:**
Port some configurations from the existing X-Ray daemon in order to implement an OT receiver for the X-Ray SDK later.

**Link to tracking Issue:**
Closes #335.

**Testing:**
Unit tests added and executed locally

**Documentation:**
README file was added

### CODE REVIEW

**Comment by hencrice** [comment]
I signed the LF CLA

**Comment by hencrice** [comment]
I signed it

**Comment by hencrice** [comment]
I'm having trouble getting the cla/linuxfoundation check to pass. I've signed the CLA and linked my GitHub user to the LinuxFoundation ID. Not sure what else to do. Could it be that I’m using the no-reply email as the author of my commits?

File: receiver/awsxrayreceiver/README.md
Comment (shengxil) [inline receiver/awsxrayreceiver/README.md:30]: Shall we clarify this is optional?

File: receiver/awsxrayreceiver/README.md
Comment (shengxil) [inline receiver/awsxrayreceiver/README.md:71]: `ECS` -> `EC2`

File: receiver/awsxrayreceiver/config.go
Comment (shengxil) [inline receiver/awsxrayreceiver/config.go:68]: Same

File: receiver/awsxrayreceiver/doc.go
Comment (shengxil) [inline receiver/awsxrayreceiver/doc.go:19]: add a new line before this

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [inline receiver/awsxrayreceiver/README.md:61]: If empty, the role of the IAM credentials will be used.

Not sure if this is the right wording but let's add something like that

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [inline receiver/awsxrayreceiver/README.md:5]: Can be separate PR but eventually need to document how to authenticate, just a link to this basically

https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [inline receiver/awsxrayreceiver/README.md:18]: `proxy_server` doesn't seem to be documented, I assumed all the fields were top-level

Comment (anuraaga) [reply receiver/awsxrayreceiver/README.md:18]: And actually, shouldn't they all be top-level? We use the X-Ray API for both segments (from endpoint) and sampling (from tcp_endpoint), all the other fields apply to using the X-Ray API, not just the proxy I believe.

File: receiver/awsxrayreceiver/README.md
Comment (hencrice) [reply receiver/awsxrayreceiver/README.md:47]: Btw I just found that the X-Ray exporter also has this field `NoVerifySSL` so that probably needs to be updated as well.

File: receiver/awsxrayreceiver/doc.go
Comment (hencrice) [reply receiver/awsxrayreceiver/doc.go:20]: lint failed. So should not have a newline. Resolved

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [reply receiver/awsxrayreceiver/README.md:19]: Ah that's right got confused by thinking about the daemon's workings.

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [reply receiver/awsxrayreceiver/README.md:47]: Thanks yeah we'll want to update exporter too shouldn't be this PR though. Looks like here you updated the code but need to update this?

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [inline receiver/awsxrayreceiver/README.md:52]: If region isn't set, what API does it hit? If it falls back to environment variable but is still required, let's document that.

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [inline receiver/awsxrayreceiver/README.md:64]: New line

File: receiver/awsxrayreceiver/factory.go
Comment (tigrannajaryan) [inline receiver/awsxrayreceiver/factory.go:93]: Since both creators return ErrDataTypeIsNotSupported it means the receiver cannot be created. What's the intent here? Is this a partial PR?

Comment (hencrice) [reply receiver/awsxrayreceiver/factory.go:93]: Correct. The interface for me to implement it

File: receiver/awsxrayreceiver/README.md
Comment (anuraaga) [inline receiver/awsxrayreceiver/README.md:22]: Isn't this

```
tls_setting:
  insecure: false
  server_name_override: ""
```
?

**Review by anuraaga — APPROVED** [review]
Just small nit, thanks

File: receiver/awsxrayreceiver/README.md
Comment (hencrice) [reply receiver/awsxrayreceiver/README.md:22]: No, the field is defined as:
```
TLSSetting configtls.TLSClientSetting `mapstructure:",squash"`
```
which is similar to how the other plugin define it. And based on:
* https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/receiver/carbonreceiver/config.go#L34 (`configmodels.ReceiverSettings` has a nested field `endpoint`)
* https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/receiver/carbonreceiver/testdata/config.yaml#L6

when a field is designated as `mapstructure:",squash"`, there will be no nested structure.

Comment (anuraaga) [reply receiver/awsxrayreceiver/README.md:22]: Ah got it - thanks! LGTM

**Comment by hencrice** [comment]
@tigrannajaryan Could you take another look when you have a minute?

**Comment by hencrice** [comment]
@tigrannajaryan Could you review? Thanks

File: receiver/awsxrayreceiver/factory.go
Comment (hencrice) [reply receiver/awsxrayreceiver/factory.go:93]: This is why I did not change the top level `go.mod` because I don't want this receiver to be included in the executable yet

File: receiver/awsxrayreceiver/README.md
Comment (tigrannajaryan) [inline receiver/awsxrayreceiver/README.md:43]: The defaults for all other receivers were recently changed to 0.0.0.0. To make it uniform it is best to use it here as well.

File: receiver/awsxrayreceiver/README.md
Comment (tigrannajaryan) [inline receiver/awsxrayreceiver/README.md:3]: Please add a note saying that this component cannot be used since it is not yet fully implemented.

File: receiver/awsxrayreceiver/factory.go
Comment (tigrannajaryan) [reply receiver/awsxrayreceiver/factory.go:93]: Sounds good. As long as it is not added to go.mod and to the list of enabled components we can treat this as an incomplete inactive code. Let's just make sure the README clearly says so.
I assume more PRs will follow soon. We will not want to keep an incomplete code around for long.

**Comment by hencrice** [comment]
@tigrannajaryan Yes I'm actually working on another PR now to enable X-Ray span ingestion using UDP. I've updated the PR based on your comments. Please take another look. Thanks!

**Comment by mtwo** [comment]
@tigrannajaryan can you re-review? @hencrice presented this at the weekly meeting today

**Comment by mtwo** [comment]
@james-bebbington will also review, as discussed in today's meeting

File: receiver/awsxrayreceiver/config.go
Comment (james-bebbington) [inline receiver/awsxrayreceiver/config.go:34]: That comment is probably overkill, given how `squash` is used throughout this repo. Also I think its fairly well implied from the name?

**Review by james-bebbington — APPROVED** [review]
Had a quick review of receiver structure, etc. LGTM!

File: receiver/awsxrayreceiver/config.go
Comment (hencrice) [reply receiver/awsxrayreceiver/config.go:34]: Thanks James. I will remove this in the followup PR I'm working on now.

Comment (hencrice) [reply receiver/awsxrayreceiver/config.go:34]: @james-bebbington Seems like the PR is still blocked on Tigran's approval. Could you remove him from the list of reviewers. I tried but couldn't do it myself. Thanks!

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Sorry for the delay.

**Pull request merged**
